### PR TITLE
Alts context

### PR DESCRIPTION
**Pull request by d-reidenbach**
@ZhenLian For Code Review

Finished Peer Attributes plumbing from Handshaker Response to ALTS Context. Included test in alts_tsi_handshaker_test

### CODE REVIEW

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 1]
```diff
@@ -334,6 +346,36 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
...
+  if(peer_identity == nullptr ) { //|| *peer_attributes_counter == nullptr) {
+    gpr_log(GPR_ERROR, "Null Peer Identity.");
+    return TSI_FAILED_PRECONDITION;
+  }
+
+  if ( ! grpc_gcp_Identity_has_attributes(const_cast<grpc_gcp_Identity*>(peer_identity))) {
+        gpr_log(GPR_ERROR, "XOXO  Failed to DEFINE PEER ATTRIBUTES ENTRY AND VALUES");
+      }
+
+  size_t iter;
```
Comment (veblush) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:362]: I guess that `iter` needs to be initialized to `UPB_MAP_BEGIN` to begin with the first element of the map.

Comment (d-reidenbach) [reply src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:362]: Thanks for finding this! Switching to that allows me to pass all my tests now

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 2]
```diff
@@ -23,6 +23,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <iostream>
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:26]: Do we really need to include this?

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 3]
```diff
@@ -251,6 +252,8 @@ static const tsi_handshaker_result_vtable result_vtable = {
     handshaker_result_create_frame_protector,
     handshaker_result_get_unused_bytes, handshaker_result_destroy};
 
+
+// LINE OF INTEREST
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:256]: You can remove this line and the empty line above.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 4]
```diff
@@ -334,6 +338,26 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
                                                  local_service_account);
   grpc_gcp_AltsContext_set_peer_rpc_versions(
       context, const_cast<grpc_gcp_RpcProtocolVersions*>(peer_rpc_version));
+
+// START OF PLUMBING
+  // grpc_gcp_HandshakerResult* ncresult = const_cast<grpc_gcp_HandshakerResult*>(hresult);
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:343]: nit: remove the comments.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 5]
```diff
@@ -334,6 +338,26 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
...
+  grpc_gcp_Identity* peer_identity = const_cast<grpc_gcp_Identity*>(identity);
+  if(peer_identity == nullptr ) {
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:345]: nit: remove the empty space after "nullptr". Ditto elsewhere.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 6]
```diff
@@ -334,6 +338,26 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
...
+    gpr_log(GPR_ERROR, "Null Peer Identity.");
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:346]: `gpr_log(GPR_ERROR, "Null peer identity in ALTS context.");`

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc [hunk 7]
```diff
@@ -672,5 +696,5 @@ alts_handshaker_client* alts_tsi_handshaker_get_client_for_testing(
   return handshaker->client;
 }
 
-}  // namespace internal
-}  // namespace grpc_core
+}
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:699]: Why do we remove these comments?

File: src/core/tsi/alts/handshaker/proto/altscontext.proto [hunk 8]
```diff
@@ -38,4 +38,5 @@ message AltsContext {
 
   // The RPC protocol versions supported by the peer.
   RpcProtocolVersions peer_rpc_versions = 6;
+
```
Comment (ZhenLian) [inline src/core/tsi/alts/handshaker/proto/altscontext.proto:41]: delete empty line.

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc (hunk 2 above)
Comment (d-reidenbach) [reply src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:26]: Good catch. Had it in here when I was trying to print but forgot to take it out

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc (hunk 5 above)
Comment (d-reidenbach) [reply src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:345]: will do!

File: src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc (hunk 7 above)
Comment (d-reidenbach) [reply src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc:699]: Think it was an accident but added them back thank you!

**Pull request closed without merging**